### PR TITLE
Port prepare-docker command from CGAP, and fix AcceptValidHeader.best_match (C4-627, C4-628, C4-776)

### PR DESCRIPTION
**Pull request by netsettler**
This branch-to-branch PR wants to build on the `containerize` branch. It includes/subsumes branches `kmp_containerize_prep` and `kmp_C4-627_containerize_prep_and_restore`. It has several effects:

* Ports the `prepare-docker` functionality from CGAP. Once that's in place, you _must_ run `prepare-docker` before the build commands will work. Its options will write appropriate `.ini` files from templates according to command line parameters you specify. Generally it's good for a new debugging build, for example, to do `prepare-docker --load-inserts` and then once a local deploy has been done, re-issue the command without `--load-inserts` in order to change the global default to not load inserts (so instead just assume they're pre-loaded and use them as such).
* Fix `AcceptValidHeaders.best_match` to use the CGAP-style support, eliminating a highly verbose deprecation warning and making us more insulated against future changes to a deprecated interface.
* Fixes [Fourfront warning about relative resource name in pkg_resources.resource_filename (C4-776)](https://hms-dbmi.atlassian.net/browse/C4-776) in `src/encoded/tests/conftest_settings.py`.
* Update health page to show more info that is relevant to an orchestrated deploy, for use in the new orchestrated `fourfront_mastertest` environment.
* Fix (again?) [Fourfront tests warn passlib.context.CryptContext.encrypt() is a deprecated name (C4-628)](https://hms-dbmi.atlassian.net/browse/C4-628), which was marked fixed but wasn't. (I think it's that it was fixed in `cgap-portal` but not `fourfront`.) This rewrites certain `.encrypt(...)` operations as `.hash(...)`. (e.g., `crypt_context.encrypt` and `EDWHash.crypt`)
* Fix a syntax problem of `'my_lab:this\is_wrong_too'` having an unesecaped backslash in `src/encoded/tests/test_types_init_collections.py`. I don't think there's a ticket.
* Prefer to use `utc_today_str` and a few others from `dcicutils` now that they have moved there.
  Remove some unused related functions (e.g., `test_compute_set_difference_one`).
* Misc PEP8

### CODE REVIEW

**Comment by netsettler** [comment]
I manually invoked GA on this commit and it succeeded. 
[CI #803](https://github.com/4dn-dcic/fourfront/actions/runs/1819554424)

File: src/encoded/__init__.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
 # from webob.cookies import JSONSerializer
 
 from .loadxl import load_all
-from .util import find_other_in_pair
+# from .util import find_other_in_pair
```
Comment (netsettler) [inline src/encoded/__init__.py]: I'm removing this commented-out code and a number of other commented-out exports.

File: src/encoded/root.py [hunk 2]
```diff
@@ -66,6 +67,35 @@ def uptime_info():
...
+    LOAD_TEST_DATA = 'load_test_data'
+    # cgap-only:
+    # METADATA_BUNDLES_BUCKET = 'metadata_bundles_bucket'
+    S3_ENCRYPT_KEY_ID = 's3_encrypt_key_id'
+    SNOVAULT_VERSION = 'snovault_version'
+    SQLALCHEMY_URL = 'sqlalchemy.url'
+    SYSTEM_BUCKET = 'system_bucket'
+    # cgap-only:
+    # TIBANNA_CWLS_BUCKET = 'tibanna_cwls_bucket'
+    # TIBANNA_OUTPUT_BUCKET = 'tibanna_output_bucket'
```
[the remark this answers is not in the document]
Comment (netsettler) [reply src/encoded/root.py]: OK.

File: src/encoded/root.py [hunk 3]
```diff
@@ -96,25 +134,31 @@ def health_page_view(request):
...
+            h.LOAD_DATA: settings.get(s.LOAD_TEST_DATA),
+            # h.METADATA_BUNDLES_BUCKET is cgap-only
+            h.NAMESPACE: settings.get(s.INDEXER_NAMESPACE),
+            h.PROCESSED_FILE_BUCKET: settings.get(s.FILE_WFOUT_BUCKET),
+            h.PROJECT_VERSION: settings.get(s.ENCODED_VERSION),
+            # h.S3_ENCRYPT_KEY_ID is cgap-only
+            h.SNOVAULT_VERSION: settings.get(s.SNOVAULT_VERSION),
+            h.SYSTEM_BUCKET: settings.get(s.SYSTEM_BUCKET),
+            # h.TIBANNA_CWLS_BUCKET is cgap-only
+            # h.TIBANNA_OUTPUT_BUCKET is cgap-only
```
[the remark this answers is not in the document]
Comment (netsettler) [reply src/encoded/root.py]: OK, I've made a change to factor those values through onto the health page. I'll need to do another round of testing, but I'm pretty sure that unless I did something stupid, it should work.

File: pyproject.toml [hunk 4]
```diff
@@ -178,6 +178,7 @@ migrate-dataset-type = "encoded.commands.migrate_dataset_type:main"
 migrate-files-aws = "encoded.commands.migrate_files_aws:main"
 owl-to-items = "encoded.commands.generate_items_from_owl:main"
 profile = "encoded.commands.profile:main"
+prepare-docker = "encoded.commands.prepare_docker:main"
```
[the remark this answers is not in the document]
Comment (netsettler) [reply pyproject.toml:181]: I'm leaving the documentation for another time, though. Tracking that as [C4-781](https://hms-dbmi.atlassian.net/browse/C4-781).

File: src/encoded/renderers.py [hunk 5]
```diff
@@ -374,7 +460,9 @@ class TransformErrorResponse(HTTPServerError):
 
     rss_limit = 256 * (1024 ** 2)  # MB
 
-    reload_process = True if registry.settings.get('reload_templates', False) else lambda proc: psutil.Process(proc.pid).memory_info().rss > rss_limit
+    reload_process = (True
+                      if registry.settings.get('reload_templates', False)
+                      else lambda proc: psutil.Process(proc.pid).memory_info().rss > rss_limit)
```
[the remark this answers is not in the document]
Comment (netsettler) [reply src/encoded/renderers.py:465]: Note well that I'm only breaking lines here for PEP8 reasons and I'm not changing anything. It's possible to make a change to the `rss_limit` here if you want, but you could as well make that change in your `containerize` PR. My preference would be to just merge this into `containerize` and then let you experiment.

File: src/encoded/renderers.py [hunk 6]
```diff
@@ -331,6 +400,22 @@ def should_transform(request, response):
...
+    # Another alternative would be for this conflict to return an error.
+    # But Fourfront thinks it should just go ahead and try the transform.
+    # Note that uncommenting this logic will require uncommenting the corresponding line in 
+    # test_should_transform that is presently marked as a bug we should fix.
+    #
+    # # If we have a 'frame' that is not None or page, force JSON, since our UI doesn't handle all various
+    # # forms of the data, just embedded/page.
+    # request_frame = request.params.get("frame", "page")
+    # if request_frame != "page":
+    #     return False
```
[the remark this answers is not in the document]
Comment (netsettler) [reply src/encoded/renderers.py]: I copied your remarks into the sources for now. Still commented-out.

**Review by willronchetti — APPROVED** [review]

**Pull request merged**